src/useCookieConsent.ts

```diff
@@ -2,49 +2,44 @@ import { useState, useEffect } from 'react';
 import * as Cookies from 'js-cookie';
 import { COOKIE_CONSENT_KEY, DEFAULT_CONSENT } from './constants';
 import {
-  CookieConsent,
-  ConsentHookStateType,
-  CookieConsentOptions,
+  UseCookieConsentType,
+  ConsentState,
   CookieWrapper,
   DidDeclineAllHandler,
   AcceptCookiesOptions,
 } from './types';
 import { allCookiesSetToValue, allPropsApproved } from './utils';
 
-export const useCookieConsent = (
-  options?: CookieConsentOptions
-): ConsentHookStateType => {
+export const useCookieConsent: UseCookieConsentType = (options) => {
   const doesStorageExist =
     typeof options !== 'undefined' && 'storage' in options;
 
   const storedConsent = doesStorageExist
     ? JSON.parse(options?.storage.getItem(COOKIE_CONSENT_KEY))
     : Cookies.getJSON(COOKIE_CONSENT_KEY);
 
-  const initialConsent: CookieConsent =
+  const initialConsent: ConsentState =
     storedConsent || options?.defaultConsent || DEFAULT_CONSENT;
 
-  const [consent, setConsent] = useState<CookieConsent>(initialConsent);
+  const [consent, setConsent] = useState<ConsentState>(initialConsent);
 
   useEffect(() => {
-    if (consent?.necessary) {
-      if (doesStorageExist) {
-        options.storage.setItem(COOKIE_CONSENT_KEY, JSON.stringify(consent));
-      } else if (
-        typeof options === 'undefined' ||
-        'consentCookieAttributes' in options
-      ) {
-        Cookies.set(
-          COOKIE_CONSENT_KEY,
-          consent,
-          options?.consentCookieAttributes
-        );
-      }
+    if (consent?.necessary && doesStorageExist) {
+      options.storage.setItem(COOKIE_CONSENT_KEY, JSON.stringify(consent));
+    } else if (
+      typeof options !== 'undefined' &&
+      'consentCookieAttributes' in options
+    ) {
```

Comment:
What happens if you don't provide options? Or if you don't provide `cookie-js` attributes through `options`? You won't save anything into cookies, but that's the whole point of this library.

Reply:
This should also be covered by the tests. There should be a test checking if the cookies are actually set, which is my fault for not implementing that.

Reply:
Sounds like a great opportunity for me to write it then!

Reply:
Yeah you can do that here, you will need to create a cookie-js mock, but it has to actually work as the browser.